src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java

```diff
@@ -124,6 +124,36 @@ public LocusIteratorByState(final Iterator<GATKRead> samIterator,
      *
      * @param samIterator the iterator of reads to process into pileups.  Reads must be ordered
      *                    according to standard coordinate-sorted BAM conventions
+     * @param downsamplingMethod information about how to downsample the reads
+     * @param includeReadsWithDeletionAtLoci Include reads with deletion at loci
+     * @param includeReadsWithNsAtLoci Include reads with Ns at loci (usually it is not needed)
+     * @param keepUniqueReadListInLIBS Keep unique read list in LIBS
+     * @param samples a complete list of samples present in the read groups for the reads coming from samIterator.
+     *                This is generally just the set of read group sample fields in the SAMFileHeader.  This
+     *                list of samples may contain a null element, and all reads without read groups will
+     *                be mapped to this null sample
```

Comment:
Is it enough with `@param header header from the reads`? I'm not sure how `ReadStateManager` use the header, because I didn't look into the class...